x-pack/legacy/plugins/maps/public/layers/sources/source_registry.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+/* eslint-disable @typescript-eslint/consistent-type-definitions */
+
+import { AbstractSourceDescriptor } from '../../../common/descriptor_types';
+import { ISource } from './source';
+
+type SourceRegistryEntry = {
+  factory: (sourceDescriptor: AbstractSourceDescriptor, inspectorAdapters: object) => ISource;
```

Comment:
not sure if we need this additional hop through a factory-function. Why not jsut use the `constructor` right away (?). It will reduce the boilerplate a little.

so just something like:

```
registerSource({
    ConstructorFunction: XYZTMSSource,
    type: EMS_XYZ
})
```
The factory-function is actually `createSourceInstance`